src/health.rs

```diff
@@ -25,7 +25,7 @@ impl HealthCheck {
     }
 
     pub fn run(&self) {
```

Comment:
In order to use `try!`, the `run` method needs to return an error. In this case, it looks like `Core::new` uses `io::Result`. So the `run` method could return something like `io::Result<()>`.